### PR TITLE
Setup of unit tests pipeline

### PR DESCRIPTION
**Pull request by Achal1607**
Added configurations for setting up the unit tests framework. It aims to improve the testing of the codebase. Code coverage, mocking and assertions libraries added.

### CODE REVIEW

File: vscode/src/test/unit/index.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import * as Mocha from 'mocha';
```
Comment (naren2605) [inline vscode/src/test/unit/index.ts:1]: we can probably follow same convention like runTest.ts which we did for integration

File: vscode/src/test/unit/jdkDownloader.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { suite, test } from 'mocha';
```
Comment (naren2605) [inline vscode/src/test/unit/jdkDownloader.unit.test.ts:1]: can we pick some small file in source code and add one meaningful test ? would that work

File: vscode/package.json [hunk 3]
```diff
@@ -757,25 +757,33 @@
...
+		"@types/sinon": "^17.0.3",
 		"@types/vscode": "^1.76.0",
 		"@types/vscode-webview": "^1.57.1",
 		"@vscode/codicons": "0.0.29",
 		"@vscode/test-electron": "^2.4.0",
+		"chai": "^4.2.0",
 		"esbuild": "^0.16.17",
 		"glob": "^7.1.6",
 		"mocha": "^9.1.2",
+		"nyc": "^17.0.0",
```
Comment (naren2605) [inline vscode/package.json:784]: could you add some minimal description on libraries we are adding in pr description?

File: patches/7271.diff [hunk 4]
```diff
@@ -12,32 +12,38 @@ index 5349a20e09..952709d10b 100644
...
 -                if ("@vscode".equals(module.getFileName().toString())) {
 -                    try (DirectoryStream<Path> sds = Files.newDirectoryStream(module)) {
 -                        for (Path sModule : sds) {
--                            checkModule(sModule, sb, tokens2Projects, project2License, bundlesDir, targetDir, externalDir, binariesList);
+-                            checkModule(sModule, sb, tokens2Projects, project2License, bundlesDir, targetDir, exter
+nalDir, binariesList);
 -                        }
 -                    }
 -                    continue;
 +                if ("eastasianwidth".equals(module.getFileName().toString())) continue;
```
Comment (naren2605) [inline patches/7271.diff:27]: are skipping liscence verfication steps for new libs added here?

Comment (Achal1607) [reply patches/7271.diff:27]: Yes

Comment (naren2605) [reply patches/7271.diff:27]: we can add required liscences in netbeans patch

**Comment by Achal1607** [comment]
Would raise another PR after refactored-code, so closing this PR.

**Pull request closed without merging**